test/distributed/org/apache/cassandra/distributed/test/metrics/RequestTimeoutTest.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Test
+    public void truncate()
+    {
+        CLUSTER.filters().verbs(Verb.TRUNCATE_REQ.id).to(2).drop();
+        Assertions.assertThatThrownBy(() -> CLUSTER.coordinator(1).execute(withKeyspace("TRUNCATE %s.tbl"), ConsistencyLevel.ALL))
+                  .is(AssertionUtils.rootCauseIs(TimeoutException.class));
+    }
+
+    @Test
+    public void casV2PrepareInsert()
+    {
+        withPaxos(Config.PaxosVariant.v2);
+
+        CLUSTER.filters().verbs(Verb.PAXOS2_PREPARE_REQ.id).to(2, 3).drop();
+        Assertions.assertThatThrownBy(() -> CLUSTER.coordinator(1).execute(withKeyspace("INSERT INTO %s.tbl (pk, v) VALUES (?, ?) IF NOT EXISTS"), ConsistencyLevel.ALL, NEXT.getAndIncrement(), NEXT.getAndIncrement()))
+                  .is(isThrowable(CasWriteTimeoutException.class));
+    }
+
+    @Test
+    public void casV2PrepareSelect()
+    {
+        withPaxos(Config.PaxosVariant.v2);
+
+        CLUSTER.filters().verbs(Verb.PAXOS2_PREPARE_REQ.id).to(2, 3).drop();
+        Assertions.assertThatThrownBy(() -> CLUSTER.coordinator(1).execute(withKeyspace("SELECT * FROM %s.tbl WHERE pk=?"), ConsistencyLevel.SERIAL, NEXT.getAndIncrement()))
+                  .is(isThrowable(ReadTimeoutException.class)); // why does write have its own type but not read?
+    }
+
+    @Test
+    public void casV2CommitInsert()
+    {
+        withPaxos(Config.PaxosVariant.v2);
+
+        CLUSTER.filters().verbs(Verb.PAXOS_COMMIT_REQ.id).to(2, 3).drop();
+        Assertions.assertThatThrownBy(() -> CLUSTER.coordinator(1).execute(withKeyspace("INSERT INTO %s.tbl (pk, v) VALUES (?, ?) IF NOT EXISTS"), ConsistencyLevel.ALL, NEXT.getAndIncrement(), NEXT.getAndIncrement()))
+                  .is(isThrowable(CasWriteTimeoutException.class));
+    }
+
+    private static void withPaxos(Config.PaxosVariant variant)
+    {
+        CLUSTER.forEach(i -> i.runOnInstance(() -> Paxos.setPaxosVariant(variant)));
+    }
+
+    private static String batch(String cql)
+    {
+        return "BEGIN " + BatchStatement.Type.UNLOGGED.name() + " BATCH\n" + cql + "\nAPPLY BATCH";
+    }
+
+    public static class BB
+    {
+        public static void install(ClassLoader cl, Integer num)
+        {
+            new ByteBuddy().rebase(Condition.Async.class)
+                           .method(named("await").and(takesArguments(2)))
+                           .intercept(MethodDelegation.to(BB.class))
+                           .make()
+                           .load(cl, ClassLoadingStrategy.Default.INJECTION);
+        }
+
+        public static boolean await(long time, TimeUnit units, @This Awaitable self, @SuperMethod Method method) throws InterruptedException, InvocationTargetException, IllegalAccessException
+        {
+            // attempt to decouple the two usages of write_request_timeout: await(write_request_timeout) and message expire
+            return (boolean) method.invoke(self, time * 10, units);
```

Comment:
Without this BB injection, the tests seem to pass. Is there a race somewhere that makes this ^ necessary?

Reply:
The race is that we use the same timeout for messages and condition wait, so the message expire system needs to happen-before the signal timeout is done.  For me, most of the time signal timeout happens-before message expires, which then means `signal=false`, which means we timeout (and were before).  Now, we need to *FORCE* message expires to happen-before in order to hit the condition this patch is fixing.

Reply:
I guess, you are calling out that the test may not be working, so we should validate that the expected branch happened... ill have to think through how to detect that... its the same exception from the caller so without changing that we don't have a clear way to know

Reply:
ok since now we have a single method to say when isTimeout=true, was able to use that to track times we returned `true` and assert that in the test.

Reply:
pushed a patch that makes the tests fail if you remove `BB`, it uses reflection to monitor `isTimeout=true` and asserts that we saw at least one (seems reads can trigger multiple).  All but CAS tests fail with it disabled, and they all pass with it disabled.